text/XXXX-externalize_private_keys.md

```diff
@@ -0,0 +1,111 @@
+- Feature Name: Externalize Private Keys
+- Start Date: <!-- (fill me in with today's date, YYYY-MM-DD) -->
+- RFC PR: <!-- (leave this empty) -->
+- Issue: <!-- (leave this empty) -->
+
+# Summary
+[summary]: #summary
+
+Separate Qri user private key from configuration, storing in the operating system's keyring, or superceeded by an envrionrment variable `QRI_PRIVATE_KEY`.
+
+# Motivation
+[motivation]: #motivation
+
+Qri users are getting themselves into un-recoverable states by manually deleting their `.qri` directory, which takes their private key with it, creating major problems when they try to re-add & change datasets. Because Qri is built on public-key infrastructure (PKI), deleting the private key has the effect of locking a user out of their own identity.
+
+More importantly, storing private keys in configuration is a security vulnerability. Coupling sensitive information with configuration renders the entire configuration sensitive.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+We inherited this bad habit of storing private keys in configuration as a "deal with it later" approach taken from IPFS, unlike Qri, IPFS identities are not "significant" in the sense that it's not tied to an identity intended for humans. Tying an identity to a private key means we need to deal with this problem much earlier.
+
+This RFC deprecates `profile.privKey` config value, moving the the private key to the user's operating system keyring. When no keyring is available, store the keyring in a seperate file called `QRI_PRIVATE_KEY` within the `.qri` directory, and present the user with a warning that they should copy this file to a safe location.
```

Comment:
I'm a big fan of using the OS keyring if available. But I really don't think we should store them within the `.qri` as a backup. The entire initial motivation is that "Qri users are ... deleting their `.qri` directory".

Instead, we should use a user owned OS-specific directory that is normally used to store application state,
`%USERDATA%/qri_key_backup/%PROFILEID%` (on Windows) or `~/Library/qri_key_backup/${PROFILEID}` (on macos) or `~/.config/qri_key_backup/${PROFILEID}` (on other POSIX systems).

Reply:
big fan of this. will change.

Reply:
Uh, I'm a bit conflicted on this. On one hand, deeply agree that `.qri` is not the best place and we should store somewhere user/os specific. On the other hand I generally don't like obscure locations for magic configs in any app. I'm still very very much inclined to suggest that `.ssh` and it's equivalent on other OS's is the best trade off. A user dealing with that dir typically should already understand the risks with it or doesn't care either way. It's a location where people expect to have their keys added. Namespacing with a dedicated directory and file name should be plenty good + opens us up to later having multiple?. The only thing that doesn't fit is that while it is a private key it's not really an ssh key.

Reply:
ok, I flip-flopped on this a bunch, and now want to leave it as-written (no extra directory) See my comment below for details.